src/Grpc.AspNetCore.Server/GrpcServiceOptions.cs

```diff
@@ -38,6 +44,35 @@ public class GrpcServiceOptions
         /// Detailed error messages include details from exceptions thrown on the server.
         /// </summary>
         public bool? EnableDetailedErrors { get; set; } = null;
+
+        /// <summary>
+        /// Gets or sets the compression algorithm used to compress messages sent from the server.
+        /// The request grpc-accept-encoding header value must contain this algorithm for it to
+        /// be used.
+        /// </summary>
+        public string ResponseCompressionAlgorithm { get; set; }
```

Comment:
I think the original names `DefaultCompressionAlgorithm` and `DefaultCompressionLevel` were ok (I think internally in the impl, calling some of the methods/fields "Response*" makes sense to distinguish between response and request logic, but there's not reason for the public API to reflect that). For C++ server, these are also called DefaultCompressionAlgorithm and DefaultCompressionLevel.  In grpc-java the "reponse" prefix is also not being used for setting the compression: e.g.
https://github.com/grpc/grpc-java/blob/master/examples/src/main/java/io/grpc/examples/experimental/CompressingHelloWorldServerPerMethod.java#L89

Reply:
In the Java example it is just called compression, but it is on the response stream.

I don't like DefaultCompressionAlgorithm because nothing ever overrides it.

Reply:
Fair enough.